packages/client/src/utils/Context.ts

```diff
@@ -12,7 +12,7 @@ export class ContextError extends Error {
     code?: string
     constructor(context: Context, message: string = '', ...args: any[]) {
         // @ts-expect-error inspectOpts not in debug types
-        super(`${context.id}: ${formatWithOptions({ ...context.debug.inspectOpts, colors: false }, message, ...args)}`)
+        super(`${context?.id}: ${formatWithOptions({ ...(context?.debug?.inspectOpts || {}), colors: false }, message, ...args)}`)
```

Comment:
it's not actually optional, but at runtime it could possibly be undefined.

This is often invoked like: 

```tyoescript
class Task implements Context {
  id = instanceId(this)
  debug = debug(this.id)
  run() {
    // …
    throw new ContextError(this, 'an error happened during run')  
  }
}
```

which ideally would spit out errors something like: `"Error: Task:1 an error happened during run"`

but if this is invoked in a manner like so:

```typescript
const task = new Task()
setTimeout(task.run)
```

then at runtime, the useful part of the error will replaced with `Uncaught TypeError: Cannot read properties of undefined (reading 'id')`. 

With the above change errors will instead look like: `"Error: undefined: an error happened during run"`. Ideally this doesn't happen at all, but if it does then the least we can do is to ensure the original error message/stack isn't thrown away or replaced.

**Key points**: Never throw away error information. Ensure errors can't throw errors.

Related to this, it's possible we should enable this typescript-eslint rule in order to help catch cases like `setTimeout(task.run)` at compile time: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unbound-method.md